pkg/http/handler/timeout.go

```diff
@@ -208,13 +208,31 @@ func (tw *timeoutWriter) WriteHeader(code int) {
 	tw.w.WriteHeader(code)
 }
 
-// tryFirstByteTimeoutAndWriteError writes an error to the responsewriter if
+// tryTimeoutAndWriteError writes an error to the responsewriter if
 // nothing has been written to the writer before. Returns whether
 // an error was written or not.
 //
 // If this writes an error, all subsequent calls to Write will
 // result in http.ErrHandlerTimeout.
-func (tw *timeoutWriter) tryFirstByteTimeoutAndWriteError(msg string) bool {
+func (tw *timeoutWriter) tryTimeoutAndWriteError(msg string) bool {
+	tw.mu.Lock()
+	defer tw.mu.Unlock()
+
+	if tw.lastWriteTime.IsZero() {
+		tw.timeoutAndWriteError(msg)
+		return true
+	}
+
+	return false
+}
+
+// tryResponseStartTimeoutAndWriteError writes an error to the responsewriter if
+// nothing has been written to the writer before. Returns whether
+// an error was written or not.
+//
+// If this writes an error, all subsequent calls to Write will
+// result in http.ErrHandlerTimeout.
```

Comment:
This is the same comment as for `tryTimeoutAndWriteError`, just double checking that's correct.

Reply:
fixed comment